src/app/components/courses-container/courses-container.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { CourseDetails } from 'src/app/models/course-details.model';
+import { CoursesDataService } from 'src/app/services/courses-data.service';
+import APP_MESSAGES from 'src/app/constants/app-messages';
+
+@Component({
+  selector: 'li-courses-container',
+  templateUrl: './courses-container.component.html',
+  styleUrls: ['./courses-container.component.scss']
+})
+export class CoursesContainerComponent implements OnInit {
+  @Input() courseHeading!: string;
+  public courseDetailsData!: CourseDetails[];
+  public currentPage: number = 0;
+  public pageSize: number = 5;
+  public courseDetailsDataToShow!: CourseDetails[];
+  public disablePrevButton: boolean = false;
+  public disableNextButton: boolean = false;
+  public filterTags: string[] = [];
+  public searchKey: string = '';
+  public noResultFoundMsg: string = APP_MESSAGES.NO_RESULT_FOUND;
+
+  constructor(private coursesDataService: CoursesDataService ) {
+  }
+
+  ngOnInit(): void {
+    this.coursesDataService.courseDetailsData.asObservable()
+                          .subscribe((courses) => { 
```

Comment:
Subscription should be destroyed properly, this would cause serious memory leak isssues, please make changes across the app